learn/ai_powered_search/rest_response_request.mdx

```diff
@@ -0,0 +1,341 @@
+      "model": "MODEL_NAME"
+    }
+  }
+}
+```
+
+In Cloudflare's case, the model is part of the API route itself and doesn't need to be specified in your `request`.
+
+### The embedding prompt
+
+The prompt corresponds to the data that the provider will use to generate your document embeddings. Its specific name changes depending on the provider you chose. In Mistral, this is the `input` field. In Cloudflare, it's called `text`.
+
+Most providers accept either a string or an array of strings. A single string will generate one request per document in your database:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": "{{text}}"
+    }
+  }
+}
+```
+
+`{{text}}` indicates Meilisearch should replace the contents of a field with your document data.
+
+An array of strings allows Meilisearch to send up to 10 documents in one request, reducing the number of API calls to the provider:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": ["{{text}}", "{{..}}"]
+    }
+  }
+}
+```
+
+When using array prompts, the first item must be `{{text}}`. If you want to send multiple documents in a single request, the second array item must be `{{..}}`.
+
+When using other tools, `input` might be called something else, like `text` or `prompt`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "text": "{{text}}"
+    }
+  }
+}
+```
+
+### Provide other request fields
+
+You may add as many fields to the `request` object as you need. Meilisearch will send them as configured when it queries the embeddings provider.
+
+For example, Mistral allows you to optionally configure an `encoding_format`. Set it by declaring this field in your embedder's `request`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": ["{{text}}", "{{..}}"],
+      "encoding_format": "float"
+    }
+  }
+}
+```
+
+## The embedding response
+
+Each provider responds to queries using its own specific structure. You must indicate where Meilisearch can find the document embeddings in the provider's response.
+
+Consult your provider's API documentation, paying attention to where it places the embeddings. Mistral, for example, returns an array of objects inside `response.data`. Each array item contains an `embedding` field,
+
+Describe the full path to the embedding array in your embedder's `response`. When you reach the field containing the embedding, use `{{embedding}}` as its value:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": "{{text}}"
+    },
+    "response": {
+      "data": {
+        "text": "{{embedding}}"
```

Comment:
```suggestion
        "embedding": "{{embedding}}"
```

If we use Mistral example here

Reply:
I decided to use Cloudflare and a generic scenario for this section since I couldn't find an embedder that did not return data in an array.